[CI] OperatorTests testPushRoundToCountToQuery failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #11984 / rocky-9_platform-support-unix](https://gradle-enterprise.elastic.co/s/rme2fa6ih4jqc)
- [elasticsearch-periodic-platform-support #11984 / rocky-9_platform-support-unix](https://gradle-enterprise.elastic.co/s/xdc4rffpayoz2)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:compute:test" --tests "org.elasticsearch.compute.OperatorTests.testPushRoundToCountToQuery" -Dtests.seed=C5923C513F3C289 -Dtests.locale=bs-Cyrl-BA -Dtests.timezone=America/Ciudad_Juarez -Druntime.java=25
```

**Applicable branches:**
9.2

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.compute.OperatorTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testPushRoundToCountToQuery'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Expected a map containing
    0: expected <0L> but was <missing>
  100: <6499L>
10000: <2297L>
```

**Issue Reasons:**
- [9.2] 2 consecutive failures in step rocky-9_platform-support-unix
- [9.2] 2 failures in test testPushRoundToCountToQuery (4.5% fail rate in 44 executions)
- [9.2] 2 failures in step rocky-9_platform-support-unix (100.0% fail rate in 2 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/OperatorTests.java
+++ b/x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/OperatorTests.java
@@ -446,10 +446,17 @@ public class OperatorTests extends MapperServiceTestCase {
             for (int p = 0; p < result.getPositionCount(); p++) {
                 actual.put(groups.getLong(p), counts.getLong(p));
             }
-            assertMap(
-                actual,
-                matchesMap().entry(0L, (long) firstGroupDocs).entry(100L, (long) secondGroupDocs).entry(10000L, (long) thirdGroupDocs)
-            );
+            var matcher = matchesMap();
+            if (firstGroupDocs > 0) {
+                matcher = matcher.entry(0L, (long) firstGroupDocs);
+            }
+            if (secondGroupDocs > 0) {
+                matcher = matcher.entry(100L, (long) secondGroupDocs);
+            }
+            if (thirdGroupDocs > 0) {
+                matcher = matcher.entry(10000L, (long) thirdGroupDocs);
+            }
+            assertMap(actual, matcher);
         };
 
         try (Directory dir = newDirectory(); RandomIndexWriter w = new RandomIndexWriter(random(), dir)) {

